server/src/main/java/com/scalar/db/server/Pauser.java

```diff
@@ -0,0 +1,71 @@
+package com.scalar.db.server;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.LongAdder;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public final class Pauser {
+  private final AtomicBoolean isPaused;
+  private final LongAdder outstandingRequestCount;
```

Comment:
It's a bit unnatural to me that `Pauser`creates `outstandingRequestCount` since `outstandingRequestCount` can/should be used without doing pause.

Maybe we should have `Stats` class (that has the number of outstanding and total success/failure counts) and it is injected through the constructor? (so that Pause requires Stats but Stats doesn't require Pauser)

Reply:
I'm thinking that we should separate the role `Stats` and `Pauser`, and each of them should have `outstandingRequestCount` independently. This is because `Pauser`'s `outstandingRequestCount` targets only the endpoints that will be paused/unpaused, but `Stats`'s one should target all of the endpoints of Scalar DB Server.

What do you think?